userdata.go

```diff
@@ -21,6 +26,28 @@ func (c *Client) GetUserData(ctx context.Context) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to decode user-data: %w", err)
 	}
+	rawBytes, err := gunzipIfNeeded(decodedBytes)
+	if err != nil {
+		return "", fmt.Errorf("failed to ungzip user-data: %w", err)
+	}
+	return string(rawBytes), nil
+}
 
-	return string(decodedBytes), nil
+// hasGzipMagicNumber checks for the gzipMagic bytes at the beginning of the source
+func hasGzipMagicNumber(source []byte) bool {
+	return bytes.HasPrefix(source, gzipMagic)
+}
+
+// gunzipIfNeeded checks for the gzip magic number and unzips the bytes if necessary,
+// otherwise it returns the original raw bytes
+func gunzipIfNeeded(raw []byte) ([]byte, error) {
```

Comment:
Should it be this rather?
```suggestion
func unzipIfNeeded(raw []byte) ([]byte, error) {`
```